peer/addrinfo_serde.go

```diff
@@ -19,19 +19,23 @@ func (pi AddrInfo) MarshalJSON() ([]byte, error) {
 
 func (pi *AddrInfo) UnmarshalJSON(b []byte) error {
 	var data map[string]interface{}
-	err := json.Unmarshal(b, &data)
-	if err != nil {
+	if err := json.Unmarshal(b, &data); err != nil {
 		return err
 	}
-	pid, err := IDB58Decode(data["ID"].(string))
-	if err != nil {
-		return err
+	if id, ok := data["ID"]; ok {
```

Comment:
Shouldn't this error if there's no ID? I'd expect ```addrInfo.UnmarshalJSON([]byte(`{"Addrs":[]}`))``` or ```addrInfo.UnmarshalJSON([]byte(`{"foo":"bar"}`))``` to error instead of returning nil. A test validating the bad path expectation would be great. No addresses is fine, but no ID isn't.

Reply:
Ok, added. PTAL.